lib/bolt/cli.rb

```diff
@@ -871,6 +871,10 @@ def assert_puppetfile_or_module_command(modules)
         raise Bolt::CLIError,
               "Unable to use command '#{old_command}' when 'modules' is configured in "\
               "bolt-project.yaml. Use '#{new_command}' instead."
+      elsif modules.nil? && options[:subcommand] == 'puppetfile'
+        msg = "Command '#{old_command}' is deprecated and will be removed in Bolt 3.0. Update your project to use "\
+              "the module management feature. For more information, see https://pup.pt/bolt-module-migrate."
```

Comment:
```suggestion
              "the module management feature. For more information, see https://pup.pt/bolt-module-migrate."
```

TODO: Create this short link once we settle on the name of the header it links to.